src/virtualenv/discovery/py_info.py

```diff
@@ -138,6 +138,21 @@ def _fast_get_system_executable(self):
                 base_executable = getattr(sys, "_base_executable", None)  # some platforms may set this to help us
                 if base_executable is not None:  # use the saved system executable if present
                     if sys.executable != base_executable:  # we know we're in a virtual environment, cannot be us
+                        # Do light validation for non-Windows virtual environments
+                        # Some installs/distributions do not provide a version-less "python" binary (PEP 394).
+                        # Try to fallback to an alternative based on version number
+                        # This should still be relatively fast since it's just a couple of `stat` calls
+                        if self.os == "posix" and (self.version_info.major, self.version_info.minor) >= (3, 11):
+                            if not os.path.exists(base_executable):
```

Comment:
One thing i'm not sure of is if we should ever return a file that doesn't actually exist. Is virtualenv better off returning `None` for `PythonInfo.system_executable`? if so, i'd probably reorder these lines

Reply:
Yes if the file does not exist we should return nothing.

Reply:
Addressed this in the latest version